### DISCUSSION TITLE
How to remove leftover junk from testing, experimenting .. that ends up in "zfs get all somefs"

### DISCUSSION THREAD

**Discussion by hput** [psy0rz/zfs_autobackup#142] [Q&A] [answered]
I've done quite a lot of experimenting, stumbling around and otherwise mismanaging zfs fs.
Running "zfs get all p1"  shows ... (snipping the normal bits)
-------       -------       -------
p1    autobackup:snap1        false                  local
p1    autobackup:s3              true                   local
p1    autobackup:snr             true                   local
p1    autobackup:sn              true                   local
p1    autobackup:nhldsl        true                   local
p1    autobackup:t1              true                   local
p1    autobackup:images1   false                  local
p1    autobackup:test1         false                  local
p1    autobackup:nhlds        true                   local
p1    absn:sn                        true                   local
-------       -------       -------
How can I remove that stuff, not just disable with `false' but remove it all together

**Comment by Scrin** [comment, marked as answer]
From the [ZFS docs](https://docs.oracle.com/cd/E19253-01/819-5461/gazup/index.html):

>You can use the zfs inherit command to clear a property value, thus causing the value to be inherited from the parent dataset.

so in your case:

```
zfs inherit autobackup:snap1 p1
```

will clear the "local" source of the property (last column in your output), causing it to inherit the value instead, and if there is nowhere to inherit from (a parent dataset or a default), it will be simply cleared

**Reply by hput** [reply]
Thank you sir.
I'm pretty sure I would not have found that... sorry if I seem lazy but finding the terms to search on is a bit of a problem when you have no idea what to look for.

**Reply by psy0rz** [reply]
the fact that you have to use inherit to delete stuff is quite hard to find indeed :)

**Reply by Scrin** [reply]
No worries, clearing or unsetting properties in particular is a bit confusing in ZFS, since there is no such thing, technically speaking. If there is no "local" value for the dataset, it will be inherited instead. However for non-standard properties "unsetting" a property can be achieved by ensuring the property is not set anywhere where it could be inherited from, ie. when there is no default and no parent dataset has it.

**Reply by hput** [reply]
running ubuntu linux 22.04  with zfs
I hit something peculiar with that technique.  I seriously doubt it is a problem with the technique it self. Well first I noticed that running 'zfs inherit autobackup:snap1 p1'  as you mentioned since its against the root of the zfs fs, then our friend 'inheritance' passes it on down the chain till all of that zfs fs is cleaned up.
I have 3 pools on the host I've mostly been playing on; (rpool, p0 and p1)  I got away clean on rpool and p0 but when I got to p1 things changed:
On first run against one of my trial tags  on p1, the prompt never came back... I let it stand for over two hours  still prompt hung up.
Of course ctl-c or ctl-d had no affect eiither.  There are less than 12  zfs fs on p1, so a low number of snaps to do.  Finally I tired of letting it set, and ran "shutdown -r" ... and I'll be damned if that didn't hang too.  The shutdown went far enough to make the host unreachable ... finally necessitating a hardware shutdown.  
I started it back up just a few minutes ago, I expected that first it would not boot or at least something else bad will have happened.  But I don't see anything so far.  The test tags are still there.  But I don't think I'll mess with that for now.  After all they don't really hurt operations at all.
